### PR TITLE
First version

### PR DESCRIPTION
**Pull request by elisabettai**
First version of UI service for spinal cord segmentation / labeling vertebrae

### CODE REVIEW

**Review by KZzizzle — APPROVED** [review]
Yeeeeaaaaahhhhh!!!

File: app/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+# TODO: Change according to your needs
+RUN apt-get update -y && \
+    apt-get install -y curl bzip2 libglib2.0-0 gcc menu libsdl1.2debian libsdl2-2.0-0 freeglut3 libgtk2.0-0 libnotify4 && \
+    rm -rf /var/lib/apt/lists 
+
+# Install sct
+RUN git clone https://github.com/spinalcordtoolbox/spinalcordtoolbox.git sct && \
+    cd sct && \
+    ./install_sct -y
```
Comment (mguidon) [inline app/Dockerfile:62]: Is this building? If so, you might get rid of a lot of intermediate data and only use the binaries

File: app/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+EXPOSE 8080
+
+ENV LC_ALL=C.UTF-8
+ENV LANG=C.UTF-8
+
+FROM base AS production
+
+# TODO: Change according to your needs
+RUN apt-get update -y && \
+    apt-get install -y curl bzip2 libglib2.0-0 gcc menu libsdl1.2debian libsdl2-2.0-0 freeglut3 libgtk2.0-0 libnotify4 && \
```
Comment (mguidon) [inline app/Dockerfile:56]: Could you benchmark the opengl behavior with software mode? If its too slow I will work on a hardware accelerated version.

**Review by mguidon — APPROVED** [review]
Only minor comments. LGTM

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
+version: '3.7'
```
Comment (GitHK) [inline docker-compose.yml]: please remove this file from the repo ad add it to the .gitignore

File: .osparc/app/metadata.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+name: "spinal-cord-toolbox-app"
+thumbnail: https://repository-images.githubusercontent.com/20361729/e268cd80-ad32-11e9-9faf-4cd1193b6144
+description: "Spinal Cord Toolbox (SCT) is a comprehensive, free and open-source set of command-line tools dedicated to the processing and analysis of spinal cord MRI data see (https://github.com/spinalcordtoolbox/spinalupport it provided (FSLeyes)."
+key: simcore/services/dynamic/spinal-cord-toolbox-app
+version: 1.0.0
+integration-version: 2.0.0
```
Comment (GitHK) [inline .osparc/app/metadata.yml]: avoids future issues with bump2version
```suggestion
integration-version: "2.0.0"
```

File: .osparc/app/runtime.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  - name: Resources
+    type: Resources
+    value:
+      # NOTE: change these limits and reservations accordingly to the service's needs
+      Limits:
+        NanoCPUs: 4000000000 # 4 CPU
+        MemoryBytes: 4294967296 # 4 GB
+      Reservations:
+        NanoCPUs: 2000000000 # 2 CPU
+        MemoryBytes: 2147483648 # 2 GB
```
Comment (GitHK) [inline .osparc/app/runtime.yml]: did you review these resources, is this correct?

**Review by GitHK — COMMENTED** [review]
Please find below some comments.

One issue:I wonder what is wrong with the GitHub workflow. does not seem to be working.

File: app/Dockerfile (hunk 1 above)
Comment (elisabettai) [reply app/Dockerfile:62]: Not sure I'm understand your comment at 100%. It builds the library in a conda environment, that generates a lot of python binaries. 
I could have a look to what can be removed.

File: .osparc/app/runtime.yml (hunk 5 above)
Comment (elisabettai) [reply .osparc/app/runtime.yml]: Yes, I've increased from the default based on a benchmark I did on my local machine. 

Do you see any problem with that? If you think it's better, I can bring the values back to default 1 CPU/1 GB RAM for reservation and limit as well.

**Pull request merged**